File: Leetcode_Python/find-the-difference-of-two-arrays.py
class Solution(object):
    def findDifference(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[List[int]]
        """
        
        set1 = set(nums1)
        set2 = set(nums2)
        result = [[], []]

        for i in set1:
            if i not in set2:
                result[0].append(i)

        for i in set2:
            if i not in set1:
                result[1].append(i)

        return result

File: Leetcode_TS/find-the-difference-of-two-arrays.ts
function findDifference(nums1: number[], nums2: number[]): number[][] {
    
    let newSet1= new Set<number>(nums1)
    let newSet2 = new Set<number>(nums2)
    let result: number[][] = [[], []]

    for (let num of newSet2){
        if (!newSet1.has(num)) {
            result[1].push(num)
        }
    }

     for (let num of newSet1){
        if (!newSet2.has(num)) {
            result[0].push(num)
        }
    }

    return result
};